app/src/ereferrals/java/org/eyeseetea/malariacare/data/remote/ElementController.java

```diff
@@ -0,0 +1,50 @@
+package org.eyeseetea.malariacare.data.remote;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+
+import com.element.utils.ElementSDKManager;
+
+import org.eyeseetea.malariacare.domain.boundary.IElementController;
+
+public class ElementController implements IElementController, ElementSDKManager.ElementActivityListener, ElementSDKManager.EnrollListener {
```

Comment:
I think the ElementController should implement an interface of ExternalVoucherRegistry or something similar. The idea would be that Element is a technology that is receiving the post-send action of a voucher, and this controller just pass the ID to element, but if later another technology want to receive that voucher ID and do a different action, it would only need to implement the same interface.